src/coreclr/jit/emitarm64.cpp

```diff
@@ -17424,7 +17424,7 @@ void emitter::emitStoreSimd12ToLclOffset(unsigned varNum, unsigned offset, regNu
     else
     {
         // Extract upper 4-bytes from data
-        regNumber tmpReg = codeGen->internalRegisters.GetSingle(tmpRegProvider);
+        regNumber tmpReg = codeGen->internalRegisters.Extract(tmpRegProvider);
```

Comment:
Interesting, this function checks whether we don't have any spare temps at all - it does the store using two shuffles without temps, otherwise, do a simpler routine with temp, but looks like in some cases we over-allocate temps for it? Given that diffs are fine I guess it's ok

Reply:
right, we may have more than 1 temps and `Extract` make sure to get the first available and remove it from the pool.